enhancements/installer/vsphere-ipi.md

```diff
@@ -0,0 +1,283 @@
+
+It will be assumed the customer's enivornment will provide properly configured
+DHCP server that is available on the layer 2 network that the vSphere networking
+is configured to.
+
+The existing supported vSphere versions will still remain supported including 6.5 and
+6.7.
+
+## Motivation
+
+VMware vSphere platform represents a significant number of OpenShift installations.
+Customers would like the option that currently includes UPI to be able to install OpenShift
+using the more automated IPI method.
+
+### Goals
+
+- Install OpenShift on vSphere using IPI method
+- The "baremetal networking" solution will be used for required OpenShift DNS and load balancing
+
+### Non-Goals
+
+- Day two replacment of internal DNS or load balancers
+- Linked clones
+
+## Proposal
+
+### UPI vs IPI
+
+Need to distinuish between vSphere UPI and IPI.  Customers that use UPI will
+will want to utlized their existing DNS and load balancing - the additional services
+are unnecessary.
+
+If the VIPs within `install-config.yaml` are undefined, nil or empty ("") then
+we will assuming UPI.  If defined and non-empty we will assume IPI.
+
+### openshift/api
+
+- Add `VSpherePlatformStatus` struct which contains `APIServerInternalIP`, `IngressIP`, and `NodeDNSIP`.
+These ip variables are used for VIPs the api server, ingress and DNS.
+
+### openshift/installer
+
+- Add additional required parameters to vSphere platform
+  - `pkg/types/vsphere/platform.go`
+
+```
+type Platform struct {
+	VCenter string `json:"vCenter"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+	Datacenter string `json:"datacenter"`
+	DefaultDatastore string `json:"defaultDatastore"`
+	Folder string `json:"folder,omitempty"`
+	Cluster string `json:"cluster,omitempty"`
+
+	APIVIP string `json:"apiVIP,omitempty"`
+	IngressVIP string `json:"ingressVIP,omitempty"`
+	DNSVIP string `json:"dnsVIP,omitempty"`
+
+    // override url for RHCOS image
+    ClusterOSImage string `json:"clusterOSImage,omitempty"`
+}
+```
+
+- Add the ability to download the OVA using the url provided in `rhcos.json`
+and upload to vsphere as a VM template. Assign `Template` variable. Based
+on the [spike](https://issues.redhat.com/browse/CORS-1295):
+  - The installer will use the baseURI and images.vmware.path from rhcos.json to download the published OVA.
+  - For disconnected, the user can provide in the install-config.yaml a full URI path (file or URL) to the OVA which the user has previously downloaded and made available.
+  - The installer will import the OVA and create a VM template and include the infraID in the name. The VM template will be placed in the root of the vSphere hierarchy.
+  - All created entities should include the infraID to allow cleanup when running 'destroy'.
+  - Out of scope:
+    - Custom placement of the VM template
+    - Custom name of the VM template
+  - `pkg/rhcos/vsphere.go`
+- Add machines and machinesets
```

Comment:
Developing for the machine API is happening here
API 
https://github.com/openshift/machine-api-operator/tree/master/pkg/apis/vsphereprovider
Controller
https://github.com/openshift/machine-api-operator/tree/master/pkg/controller/vsphere

We'll start by supporting only vCenter interactions (no raw esxi) and only supporting cloning templates on "creation". We'll extend this behaviour later on only if there's a product requirement.

We are assuming the existence of a secret with a user and password keys used to login against the vCenter https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/machine_scope.go#L57